pkg/spoke/addon/registration_controller.go

```diff
@@ -74,11 +79,44 @@ func NewAddOnRegistrationController(
 			},
 			hubAddOnInformers.Informer()).
 		WithSync(c.sync).
+		ResyncEvery(10*time.Minute).
 		ToController("AddOnRegistrationController", recorder)
 }
 
 func (c *addOnRegistrationController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	addOnName := syncCtx.QueueKey()
+	queueKey := syncCtx.QueueKey()
+	if queueKey != factory.DefaultQueueKey {
+		// sync a particular addOn
+		return c.syncAddOn(ctx, syncCtx, queueKey)
+	}
+
+	// handle resync
+	addOns, err := c.hubAddOnLister.ManagedClusterAddOns(c.clusterName).List(labels.Everything())
```

Comment:
i think we get all existing addons from addOnRegistrationConfigs,  so we can compare with current lister

Reply:
Code updated to sync between cached registration configs and ManagedClusterAddOns in informer cache.